### PR TITLE
Build hashsum, join, ln, ls, mkdir, mktemp, nl, ptx, sort on Redox

### PR DESCRIPTION
**Pull request by ids1024**

### CODE REVIEW

**Comment by jackpot51** [comment]
I jumped the gun on merging it - sorry about that @Arcterus

**Comment by jackpot51** [comment]
I reverted it. It looks like the tests failed @ids1024 - can you look into that?

**Comment by Arcterus** [comment]
@jackpot51 no problem.  Was there something wrong with the commit?  It looks fine to me, but perhaps I’m missing something.

File: src/mkdir/mkdir.rs [hunk 1]
```diff
@@ -135,20 +135,18 @@ fn mkdir(path: &Path, mode: u16, verbose: bool) -> i32 {
...
+        use std::os::unix::fs::{PermissionsExt};
 
-        let directory =
-            CString::new(path.as_os_str().to_str().unwrap()).unwrap_or_else(|e| crash!(1, "{}", e));
-        let mode = mode as libc::mode_t;
+        let mode = Permissions::from_mode(mode as u32);
 
-        if unsafe { libc::chmod(directory.as_ptr(), mode) } != 0 {
+        if let Err(err) = set_permissions(path, mode) {
             show_info!(
```
Comment (Arcterus) [inline src/mkdir/mkdir.rs]: This should actually be `show_error!` (and then the `error {}` should be `({})`).

**Comment by ids1024** [comment]
There was a test failure on OS X in `chmod`, which I didn't change. Merging also cancelled the Travis build; I tried to restart it, but that doesn't seem to work with a PR that has already been built.

CI seems to be passing on my own repo (https://travis-ci.org/ids1024/uutils/builds/353585248), so I think that failure was spurious.

**Comment by Arcterus** [comment]
Ah alright.  Which test failed specifically?  We might want to look at it to ensure there aren’t any issues with it.

**Comment by ids1024** [comment]
> Which test failed specifically? 

It was `test_chmod_ugoa`, I think.

I've opened a new PR, with that error code changed (#1161).

**Comment by Arcterus** [comment]
Okay, it seems fine from briefly looking at it.  I'll have to investigate more thoroughly if it happens again.

**Pull request merged**